src/lib.rs

```diff
@@ -1,5 +1,102 @@
+use std::marker::PhantomData;
+
 #[cfg(target_os = "macos")]
-pub mod macos;
+mod macos;
+
+#[cfg(target_os = "android")]
+mod android;
+
+#[cfg(target_os = "windows")]
+mod windows;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("Unsupported platform")]
+    Unsupported,
+    /// Permissions to access location / device are not allowed, and cannot be accessed until the
+    /// user manually intervenes.
+    #[error("Permissions denied")]
+    PermissionsDenied,
+    #[cfg(target_os = "windows")]
+    #[error("{0}")]
+    Windows(#[from] windows_result::Error),
+}
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+pub struct Getter {
+    inner: Inner,
+    _phantom: PhantomData<*mut ()>,
```

Comment:
what does this phantom data achieve? I haven't seen it outside of holding references or generic types. Is this just to prevent it from being thread safe? and if so, can we achieve that with marker types/is there a more readable way to achieve this?

Reply:
Yeah `*mut` is `!Send` and `!Sync`, so putting it inside `PhantomData` consumes no space but prevents `Getter` from auto deriving the send and sync marker traits. There isnt a cleaner way to do this that I'm aware of

https://doc.rust-lang.org/nomicon/phantom-data.html